internal/dag/builder_test.go

```diff
@@ -83,6 +83,11 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 						MatchLabels: map[string]string{
 							"app": "contour",
 						},
+						MatchExpressions: []metav1.LabelSelectorRequirement{{
```

Comment:
Does this mean we're not testing the "no selectors" case explicitly?

Reply:
Yup we're not, I'll add a test for that. =)